### PR TITLE
feat(SeaTable Node): Refactor, add new triggers and actions

### PR DESCRIPTION
**Pull request by christophdb**
# Complete rework of the SeaTable Node.

- Adding two new triggers and 12 new actions.
- Actions use now versioning
- cleaner code structure
- already published and tested as community node (n8n-nodes-seatable)

### CODE REVIEW

**Comment by Joffcom** [comment]
Cheers @christophdb

**Comment by Joffcom** [comment]
@christophdb Looks like there is still a lint issue that needs to be resolved, It may be worth removing some of the comments that are not needed as well.

**Comment by christophdb** [comment]
@Joffcom I am sorry, that my PR still had some issues. These "type check errors" were not raised by my VS Code. Now they should be fixed. Can you please try again? Thanks

**Comment by Joffcom** [comment]
Hey @christophdb,

It looks like it is still reporting a bunch of lint issues, I have this pencilled in for review tomorrow morning.

**Comment by netroy** [comment]
I wanted to help fix this up, but unfortunately this PR is not editable by maintainers.
Can you please cherry-pick the last commits from [here](https://github.com/n8n-io/n8n/compare/master...seatable_node_rework2), so that we can get the review started on this 🙏🏽

**Comment by netroy** [comment]
For any future PRs please leave this checkbox enabled 🙏🏽 
image

So that we can work with your PRs
image

**Comment by christophdb** [comment]
@netroy: I cherry-picked the commits. Please tell me if I can do something else.

**Comment by Joffcom** [comment]
Overall I am happy with this, I will need to update our documentation when we merge this but I would like to take a moment to thank you @christophdb for taking on the work of making sure we are using the latest and greatest options that Seatable has to offer.

**Comment by janober** [comment]
Thanks a lot, @christophdb and @Joffcom, for getting it done!

**Comment by MehrCurry** [comment]
I am more than happy that this MR came back to life. A big thanks to all who made it possible.

**Comment by michael-radency** [comment]
Hello @christophdb, thanks for your contribution
In the latest overhauls we moved from creating folder with three separate files for each operation to using single file `<operationName>.operation.ts` with two exports: description and execute.
Could you please consider updating your PR referring to v2 implementation of [Microsoft Outlook](https://github.com/n8n-io/n8n/tree/master/packages/nodes-base/nodes/Microsoft/Outlook/v2/actions) and [Airtable](https://github.com/n8n-io/n8n/tree/master/packages/nodes-base/nodes/Airtable/v2/actions/record) nodes

**Comment by christophdb** [comment]
@michael-radency I updated the action folder with all operations as requested.

**Comment by michael-radency** [comment]
> @michael-radency I updated the action folder with all operations as requested.

Thank you @christophdb , I'll try to review this PR today

**Comment by christophdb** [comment]
@michael-radency Please don't merge now. My changes broke one of the loadOptionsDependsOn.

**Comment by christophdb** [comment]
I found my copy-and-paste error and fixed it. Now you can merge it.

**Comment by christophdb** [comment]
@michael-radency @Joffcom, is there anything else I can do at the moment?

**Comment by michael-radency** [comment]
Hello @christophdb, code changes LGTM, but this overhaul requires more testing, there are some regression in v1 and I did encountered issues in v2

in v1 loaders are broken
image
in v2 in some operations data not loaded, also ID shouldn't be appended to name
image
loaded in search though
image

File: packages/nodes-base/nodes/SeaTable/v2/actions/router.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+			} else if (seatable.resource === 'link') {
+				responseData = await link[seatable.operation].execute.call(this, i);
+			} else if (seatable.resource === 'asset') {
+				responseData = await asset[seatable.operation].execute.call(this, i);
+			}
+
+			const executionData = this.helpers.constructExecutionMetaData(
+				this.helpers.returnJsonArray(responseData),
+				{ itemData: { item: i } },
+			);
```
Comment (michael-radency) [inline packages/nodes-base/nodes/SeaTable/v2/actions/router.ts:38]: `responseData` would be` INodeExecutionData[]` so can be simplified to
```
			const executionData = this.helpers.constructExecutionMetaData(responseData, {
				itemData: { item: i },
			});
```

**Comment by christophdb** [comment]
I will have a look at these issues shortly.
@michael-radency: Just one question: how can I force n8n to create a node using the v1 version?

**Comment by christophdb** [comment]
Hey @michael-radency,
thanks for your feedback. Here is what I did:

- broken v1 loader: 
fixed with commit 7d4f162

- in v2 in some operations data not loaded (found in asset - upload): 
the error message just says, that a column has to be selected. But the options are loaded correctly and available. As soon as a value is selected, everything is fine. What change would you expect here?
image

- ID shouldn't be appended to name (found in asset - upload): 
I would like to keep this behaviour. From a technical perspective, I have to pass the ID:
image
But offering just the id as select options is not a good idea, because it is difficult for the user to determine the id from the web interface. On the other hand it is not a good idea to show only the name of the first column, because multiple rows can contain the name value. 
Therefore, I would like to keep the behaviour it like this:
image

- loaded in search though (row - search): 
I don't have this issue. The loading is working fine. Error is just: value is required. What change would you expect here?
image

- simplified router.ts:
I improved the code like you suggested.

Best regards
Christoph

**Comment by christophdb** [comment]
Dear @netroy or @michael-radency,
Is there anything I can do?

**Comment by christophdb** [comment]
Dear @netroy or @michael-radency,
sorry to bother you. Any updates on this or something that I can do?
Best regards
Christoph

**Comment by Joffcom** [comment]
Hey @christophdb,

I am going to pick this one back up again during my next review session on Thursday morning.

**Comment by Joffcom** [comment]
@christophdb it looks like there are some conflicts that need to be resolved again, I want to get this one merged in this quarter.

**Comment by netroy** [comment]
Since maintainers can't edit this PR, can you please cherry-pick the last 2 commits from [here](https://github.com/n8n-io/n8n/commits/seatable_node_rework2/) 🙏🏽

**Comment by Joffcom** [comment]
Hey @christophdb,

Because we can't edit this PR and there are a lot of conflicts I have created a new PR with your changes here: https://github.com/n8n-io/n8n/pull/11431

This should allow us to work quickly to get this merged. I am going to close this one for now in favour of that one.

**Pull request closed without merging**